### PR TITLE
Migrate Microsoft.Bot.Builder.AI.Luis.Tests to xUnit

### PR DESCRIPTION
**Pull request by sw-joelmut**
Fixes # 4086

## Description
This PR migrates all [Microsoft.Bot.Builder.AI.Luis](https://github.com/microsoft/botbuilder-dotnet/tree/main/tests/Microsoft.Bot.Builder.AI.LUIS.Tests) tests from **MSTest** to **xUnit**.

## Specific Changes
- Replaced `MSTest` packages with `xUnit` in the `.csproj`.
- Changed `[TestMethod]` to `[Fact]`.
- Changed `[TestCategory]` to `[Trait]`.
- Changed `[ExpectedException]` to `Assert.ThrowsAsync`.
- Changed `Assert.ThrowsException` to `Assert.Throws`.
- Changed `Assert.AreEqual` to `Assert.Equal`.
- Changed `Assert.IsNotNull` to `Assert.NotNull`.
- Changed `Assert.IsNull` to `Assert.Null`.
- Changed `Assert.Fail` to `throw new XunitException`.
- Added `[assembly: TestFramework("Microsoft.Bot.Builder.Startup", "Microsoft.Bot.Builder.AI.Luis.Tests")]` to the `Startup` class to replace `[AssemblyInitialize]`.
- Implemented `LuisAdaptiveRecognizerFixture` class to share context among tests.
- Improved tests Assertions.

## Testing
In the following image there are the passing tests.
image

### CODE REVIEW

File: tests/Microsoft.Bot.Builder.AI.LUIS.Tests/LuisV3OracleTests.cs [hunk 1]
```diff
@@ -80,19 +79,20 @@ public void EmptyEndpoint()
...
+        [Fact]
         public void LuisRecognizer_NullLuisAppArg()
         {
-            var recognizerWithNullLuisApplication = new LuisRecognizer(new LuisRecognizerOptionsV3(null));
-            Assert.Fail();
-            Assert.IsNotNull(recognizerWithNullLuisApplication);
+            Assert.Throws<ArgumentNullException>(() =>
+            {
+                var recognizerWithNullLuisApplication = new LuisRecognizer(new LuisRecognizerOptionsV3(null));
+                Assert.NotNull(recognizerWithNullLuisApplication);
```
Comment (ceciliaavila) [inline tests/Microsoft.Bot.Builder.AI.LUIS.Tests/LuisV3OracleTests.cs:91]: Is this line executed?

File: tests/Microsoft.Bot.Builder.AI.LUIS.Tests/LuisV3OracleTests.cs [hunk 2]
```diff
@@ -205,126 +134,129 @@ public async Task TraceActivity()
...
+        [Fact]
         public async Task GeoPeopleOrdinal() => await TestJson<RecognizerResult>("GeoPeopleOrdinal.json");
 
-        [TestMethod]
+        [Fact]
         public async Task Minimal() => await TestJson<RecognizerResult>("Minimal.json");
 
         // TODO: This is disabled until the bug requiring instance data for geo is fixed.
-        // [TestMethod]
+        // [Fact]
```
Comment (ceciliaavila) [inline tests/Microsoft.Bot.Builder.AI.LUIS.Tests/LuisV3OracleTests.cs:194]: Maybe we can skip this test with the TODO as the reason.

**Review by ceciliaavila — COMMENTED** [review]
Just a suggestion and a question.

File: tests/Microsoft.Bot.Builder.AI.LUIS.Tests/LuisV3OracleTests.cs (hunk 1 above)
Comment (sw-joelmut) [reply tests/Microsoft.Bot.Builder.AI.LUIS.Tests/LuisV3OracleTests.cs:91]: It is not, do you think we should remove it? Maybe it was needed as a precaution for MSTests tests, I think if I could remember, there are more potential cases like this one.

Comment (ceciliaavila) [reply tests/Microsoft.Bot.Builder.AI.LUIS.Tests/LuisV3OracleTests.cs:91]: Yes, there're a few more cases like this, I think we should remove those unused asserts.

File: tests/Microsoft.Bot.Builder.AI.LUIS.Tests/LuisV3OracleTests.cs (hunk 2 above)
Comment (sw-joelmut) [reply tests/Microsoft.Bot.Builder.AI.LUIS.Tests/LuisV3OracleTests.cs:194]: We did the same for this PR https://github.com/southworks/botbuilder-dotnet/pull/314/files#diff-ab9c3185422704bbf99a9cb1aff1a056R216

Comment (ceciliaavila) [reply tests/Microsoft.Bot.Builder.AI.LUIS.Tests/LuisV3OracleTests.cs:194]: Yeah, that sounded familiar to me. Ok, let's keep it like this to have consistency.

**Review by ceciliaavila — APPROVED** [review]

**Review by santgr11 — APPROVED** [review]

**Review by denscollo — APPROVED** [review]

**Comment by denscollo** [comment]
Promoted

**Pull request closed without merging**